anime_downloader/animeinfo.py

```diff
@@ -0,0 +1,216 @@
+        Higher number = more similar.
+    """
+    def __init__(self, AnimeInfo, SearchResult, ratio = 100):
+        self.AnimeInfo = AnimeInfo
+        self.SearchResult = SearchResult
+        self.ratio = ratio
+
+
+def search_mal(query):
+
+    def search(query):
+        soup = helpers.soupify(helpers.get('https://myanimelist.net/anime.php', params = {'q':query}))
+        search_results = soup.select("a.hoverinfo_trigger.fw-b.fl-l")
+        return [SearchResult(
+            url = i.get('href'),
+            title = i.select('strong')[0].text
+            ) for i in search_results]
+
+
+    def scrape_metadata(url):
+        soup = helpers.soupify(helpers.get(url))
+        """
+        info_dict contains something like this: [{
+        'url': 'https://myanimelist.net/anime/37779/Yakusoku_no_Neverland',
+        'title': 'The Promised Neverland',
+        'jp_title': '約束のネバーランド'
+        },{
+        'url': 'https://myanimelist.net/anime/39617/Yakusoku_no_Neverland_2nd_Season',
+        'title': 'The Promised Neverland 2nd Season',
+        'jp_title': '約束のネバーランド 第2期'}]
+        """
+        info_dict = {
+            'url':url
+        }
+
+        # Maps specified info in sidebar to variables in info_dict
+        name_dict = {
+        'Japanese:':'jp_title',
+        'English:':'title',
+        'synonyms:':'synonyms',
+        'Episodes:':'episodes'
+        }
+        info = soup.select('span.dark_text')
+        extra_info = [i.parent.text.strip() for i in info]
+        for i in extra_info:
+            text = i.replace('\n','').strip()
+            for j in name_dict:
+                if text.startswith(j):
+                    info_dict[name_dict[j]] = text[len(j):].strip()
+
+        # Backup name if no English name isn't registered in sidebar
+        if not info_dict.get('title'):
+            name = soup.select('span[itemprop=name]')
+            info_dict['title'] = name[0].text if name else None
+
+        # Always sets episodes
+        if not info_dict.get('episodes') or info_dict.get('episodes') == 'Unknown':
+            info_dict['episodes'] = 0
+
+        # TODO error message when this stuff is not correctly scraped
+        # Can happen if MAL is down or something similar
+        return AnimeInfo(url = info_dict['url'], title = info_dict.get('title'),
+                jp_title = info_dict.get('jp_title'), episodes = int(info_dict['episodes']))
+
+    search_results = search(query)
+    season_info = []
+    # Max 10 results
+    for i in range(min(len(search_results), 10)):
+        anime_info = scrape_metadata(search_results[i].url)
+        if anime_info.episodes:
+            season_info.append(anime_info)
+
+    # Code below uses the first result to compare
+    #season_info = [scrape_metadata(search_results[0].url)] 
+    #return season_info
+
+    # Prompts the user for selection
+    return primitive_search(season_info)
+
+
+def search_anilist(query):
```

Comment:
What is the use of anilist search if it's doing the exact same thing as MAL?

Reply:
Will remove MAL